pathfinderlib/main.py

```diff
@@ -65,32 +282,76 @@ def main(argv=None):
         type=argparse.FileType("w"),
         help="Summary file")
     parser.add_argument(
-        "--output-file", required=True,
-        help="output file")
+        "--output-dir", required=True,
+        help="output directory")
     parser.add_argument(
         "--config-file",
         type=argparse.FileType("r"),
         help="JSON configuration file")
     args = parser.parse_args(argv)
 
-    config = default_config.copy()
-    if args.config_file:
-        user_config = json.load(args.config_file)
-        config.update(user_config)
+    config = get_config(args.config_file)
 
     fwd_fp = args.forward_reads.name
     rev_fp = args.reverse_reads.name
     args.forward_reads.close()
     args.reverse_reads.close()
 
-    app = Humann(config)
-    app.run(fwd_fp, rev_fp, args.output_file)
-    save_summary(args.summary_file, config)
+    if not os.path.exists(args.output_dir):
+        os.mkdir(args.output_dir)
+
+    searchApp = Alignment(config)
+    alignment_R1_fp = searchApp.run(fwd_fp, args.output_dir)
+    alignment_R2_fp = searchApp.run(rev_fp, args.output_dir)
+
+    assignerApp = Assignment(config)
+    summary_R1 = assignerApp.run(alignment_R1_fp, args.output_dir)
+    summary_R2 = assignerApp.run(alignment_R2_fp, args.output_dir)
+
+    summary_R1.update({'num_sequences': count_seq(fwd_fp)})
+    summary_R2.update({'num_sequences': count_seq(rev_fp)})
+    
+    save_summary(args.summary_file, config, {'R1':summary_R1, 'R2':summary_R2})
 
-def save_summary(f, config):
+def save_summary(f, config, data):
     result = {
         "program": "PathFinder",
         "version": __version__,
-        "config": config
+        "config": config,
+        "data":data
         }
-    json.dump(result, f)    
+    json.dump(result, f)
+
+def count_seq(R):
+    "Returns the number of sequences in the query file. Assumes fastq!!"
+    stdout = subprocess.check_output(['wc', '-l', R], stderr=subprocess.STDOUT)
+    return int(stdout.split(' ', 1)[0])/4
+                        
+def make_index_main(argv=None):
+    p = argparse.ArgumentParser()
+    p.add_argument(
+        "--config-file",
+        type=argparse.FileType("r"),
+        help="JSON configuration file")
+    args = p.parse_args(argv)
+
+    config = get_config(args.config_file)
+        
+    searchApp = Alignment(config)
+    if not searchApp.index_exists():
+        searchApp.make_index()
+
+    assignerApp = Assignment(config)
+    if not assignerApp.index_exists():
+        assignerApp.make_index()
```

Comment:
This is really nice, may be changing my opinion on how/where we keep the logic for indexing databases.  Maybe each package should know how to index its own data.  I like this -- let's keep it!